video/debate/quarto_mexca_performance_video.qmd

```diff
@@ -0,0 +1,343 @@
+
+From this dataset, we have created 10 videos lasting from 00:24 to 2:17 and featuring from 1 to 10 individuals.
+
+This is an example of a video featuring two individuals:
+
+```{r}
+embedr::embed_video('total/2_speaker_total.mp4')
+```
+
+The videos are build by selecting and concatenating shots of the most frequent individuals from 1 to 10, as reported by a human annotator in the reference file loaded below.
+
+```{r}
+camera_shots_annotation <- read.csv("camera_shots_annotation.csv", header = T)
+```
+
+Each shot is composed of a variable number of frames. The shots are all 25fps, thus a frame is 1/25 = `r 1/25`
+
+```{r}
+camera_shots_annotation |>
+  dplyr::group_by(filename) |>
+  dplyr::summarise(frames = mean(frames)) |>
+  dplyr::mutate(seconds = frames * 0.04) 
+  
+```
+
+The majority of the shots includes annotations of one single person at a time, and of max 3 individuals at a time:
+
+```{r}
+camera_shots_annotation$n_faces <- rowSums(!is.na(camera_shots_annotation[,c("face_1", "face_2", "face_3")]))
+
+table(camera_shots_annotation$n_faces)
+```
+
+We can answer Q1 by evaluating mexca's performance on videos featuring (1) individuals who are alone in the frame, or (2) are together with other people. From the annotation structure above, we know however that for (2) we can evaluate the performance for min = 2 to max = 3 people at a time only.
+
+Among all the individuals, we have excluded individuals who were not frequently annotated. This led us to have 10 people.
+
+```{r}
+sort(table(camera_shots_annotation$face_1), decreasing = T)
+
+```
+
+## Mexca's output
+
+We read in the relevant columns in mexca output which allow us to compute performance for face detection and identification.
+
+```{r}
+#| warning: false
+output_files <- list.files('output/with_n_clusters/')
+
+shot_df <- data.frame() 
+
+for(file in output_files){
+  ded21 <- read.csv(paste0("output/with_n_clusters/",file),stringsAsFactors = T, header = T, dec = ".", na.strings = ' ') 
+  ded21 <- ded21[,c("frame", "time","face_prob", "face_landmarks", "face_aus", "face_id")]
+  ded21 <- unpack_face_landmarks(ded21)
+  ded21 <- unpack_face_aus(ded21)
+  ded21 <- ded21 |> dplyr::filter_all(dplyr::any_vars(!is.na(.)))
+  ded21$file <- as.factor(file); 
+  
+  shot_df <- dplyr::bind_rows(shot_df, ded21)
+}
+
+shot_df <- unique(shot_df)
+
+head(shot_df[,c("frame","time","face_id","face_prob","file")])
+
+```
+
+We add to mexca's output the column `shot_id` and `filename` indicating the shots that made each video in the file/`video_name` column. Moreover, we also add in the column `n` the order in which the shots were concatenated. This info is useful in order to understand who was displayed when.
+
+```{r}
+input_files <- c(list.files('total', full.names = T), list.files('contemporary', full.names = T))
```

Comment:
In this chunk, it does not seem to find the `input_files`. Maybe this is just a path issue?

Reply:
Nope! my mistake, It was supposed to load a folder that was not present. I loaded it now

Reply:
It works now, great!